### PR TITLE
[Mellanox] Enhance Python3 support for platform API

### PR DESCRIPTION
**Pull request by stephenxs**
#### Why I did it
Enhance the Python3 support for platform API.
Originally, some platform APIs call SDK API which didn't support Python 3. Now the Python 3 APIs have been supported in SDK 4.4.3XXX, Python3 is completely supported by platform API

Signed-off-by: Stephen Sun <stephens@nvidia.com>

#### How I did it
- Start all platform daemons from python3
- Remove `#/usr/bin/env python` at the beginning of each platform API file as the platform API won't be started as daemons but be imported from other daemons.
- Adjust SDK API calls accordingly

#### How to verify it
- Manually test
- Run regression platform test

#### Which release branch to backport (provide reason below if selected)

- [ ] 201811
- [ ] 201911
- [ ] 202006
- [x] 202012

### CODE REVIEW

File: platform/mellanox/mlnx-platform-api/sonic_platform/sfp.py [hunk 1]
```diff
@@ -1493,9 +1491,12 @@ def mgmt_phy_mod_pwr_attr_get(cls, power_attr_type, sdk_handle, sdk_index):
         sx_mgmt_phy_mod_pwr_attr = sx_mgmt_phy_mod_pwr_attr_t()
```
Comment (keboliu) [inline platform/mellanox/mlnx-platform-api/sonic_platform/sfp.py:1491]: So this struct name not changed with the new function?

File: platform/mellanox/mlnx-platform-api/sonic_platform/sfp_event.py [hunk 2]
```diff
@@ -1,4 +1,3 @@
-#!/usr/bin/env python
```
Comment (keboliu) [inline platform/mellanox/mlnx-platform-api/sonic_platform/sfp_event.py:1]: will SDK API called inside this function will also return the module id as an extra?

File: platform/mellanox/mlnx-platform-api/sonic_platform/sfp.py (hunk 1 above)
Comment (stephenxs) [reply platform/mellanox/mlnx-platform-api/sonic_platform/sfp.py:1491]: No.

File: platform/mellanox/mlnx-platform-api/sonic_platform/sfp_event.py (hunk 2 above)
Comment (stephenxs) [reply platform/mellanox/mlnx-platform-api/sonic_platform/sfp_event.py:1]: Yes. But we don't need to handle them

**Comment by stephenxs** [comment]
Depends on Azure#7848

**Comment by stephenxs** [comment]
Depends on Azure#7848

**Review by jleveque — APPROVED** [review]

**Review by liat-grozovik — APPROVED** [review]

**Comment by qiluo-msft** [comment]
This PR could not cleanly cherry-pick to 202012, please submit another PR

**Comment by stephenxs** [comment]
> This PR could not cleanly cherry-pick to 202012, please submit another PR

Thanks. I'll check whether there are some depended PRs that can be cherry-picked and will create one for 202012 if not.

**Comment by stephenxs** [comment]
> This PR could not cleanly cherry-pick to 202012, please submit another PR

PR Azure#7910 opened for back porting it to 202012.

**Pull request merged**